src/power_grid_model/core/power_grid_model.py

```diff
@@ -355,56 +363,57 @@ def calculate_state_estimation(
         output_component_types: Optional[Union[Set[str], List[str]]] = None,
         continue_on_batch_error: bool = False,
     ) -> Dict[str, np.ndarray]:
-        """
-        Calculate state estimation once with the current model attributes.
-        Or calculate in batch with the given update dataset in batch
+        """This is an example of a module level function.
 
-        Args:
-            symmetric:
-                True: three-phase symmetric calculation, even for asymmetric loads/generations
-                False: three-phase asymmetric calculation
-            error_tolerance:
-                error tolerance for voltage in p.u., only applicable when iterative=True
-            max_iterations:
-                maximum number of iterations, only applicable when iterative=True
-            calculation_method: an enumeration
-                iterative_linear: use iterative linear method
-            update_data:
-                None: calculate state estimation once with the current model attributes
-                A dictionary for batch calculation with batch update
-                    key: component type name to be updated in batch
-                    value:
-                        a 2D numpy structured array for homogeneous update batch
-                            Dimension 0: each batch
-                            Dimension 1: each updated element per batch for this component type
-                        **or**
-                        a dictionary containing two keys, for inhomogeneous update batch
-                            indptr: a 1D integer numpy array with length n_batch + 1
-                                given batch number k, the update array for this batch is
-                                data[indptr[k]:indptr[k + 1]]
-                                This is the concept of compressed sparse structure
-                                https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html
-                            data: 1D numpy structured array in flat
-            threading:
-                only applicable for batch calculation
-                < 0 sequential
-                = 0 parallel, use number of hardware threads
-                > 0 specify number of parallel threads
-            output_component_types: list or set of component types you want to be present in the output dict.
-                By default all component types will be in the output
-            continue_on_batch_error: if the program continues (instead of throwing error) if some scenarios fails
+        Function parameters should be documented in the ``Args`` section. The name
+        of each parameter is required. The type and description of each parameter
+        is optional, but should be included if not obvious.
 
+        If \*args or \*\*kwargs are accepted,
```

Comment:
as you can see with this failed action: https://github.com/PowerGridModel/power-grid-model/actions/runs/5419899136/jobs/9853457656

```
/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/site-packages/power_grid_model/core/power_grid_model.py:372:11: W1401: Anomalous backslash in string: '\*'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

we'll have to talk about this one. `r`-strings may mess up this PR as well. https://peps.python.org/pep-0287/#docstring-significant-features says that you shouldn't need that, but i don't  know what the google docstring style format and numpy docstring dictate (or what the converters require, for all that matters).

I really hope that you won't need these `\` and if you do, this provides valuable input for the discussion about which style to choose

Reply:
maybe this helps for the numpydoc style:

When documenting variable length positional, or keyword arguments, leave the leading star(s) in front of the name:
(taken from https://numpydoc.readthedocs.io/en/latest/format.html#parameters)

> *args : tuple
>     Additional arguments should be passed as keyword arguments
> **kwargs : dict, optional
>     Extra arguments to `metric`: refer to each metric documentation for a
>     list of all possible arguments.

Reply:
see also this thread on the google docstring style: https://stackoverflow.com/questions/1137161/what-is-the-correct-way-to-document-a-kwargs-parameter

They do use escape `\` characters and this may be an argument against google docstring